### PR TITLE
Add scripts starting mongod requiring SSL

### PR DESCRIPTION
**Pull request by metamben**
These scripts can be used by providing --entrypoint /bin/mongo-with-ssl.sh in the docker run command.

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
This will most likely not be needed.
All tests ran successfully in https://github.com/metabase/metabase/pull/27147

File: dbs/mongo/4.0/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,8 @@ COPY data/load.sh /docker-entrypoint-initdb.d/
 # in the image so that they can be pulled from them for local testing.
 COPY certificates/metabase.* certificates/metamongo.* certificates/metaca.* /etc/mongo/
 
+COPY 4.0/mongo-with-ssl.sh /bin/
```
Comment (camsaul) [inline dbs/mongo/4.0/Dockerfile:16]: Probably worth pointing out that we don't actually support MongoDB 4.0 any more it's EOL'ed upstream https://www.mongodb.com/support-policy/lifecycles

We should probably stop testing against it. 4.2 is the minimum version we support now

**Review by camsaul — APPROVED** [review]
While we may not **need** this PR it doesn't seem like it's hurting anything to me. So I'm give you a 👍 if you still think there is a use case for it.

If the use case is just for local development it should probably go in  https://github.com/metabase/dev-scripts instead since this repo says

> This repository contains: Docker images with pre-populated data for integration testing

And I don't think your Mongo database has any pre-populated data, right? The stuff in `dev-scripts` are for running general data warehouse DBs to develop against where the tests create the data themselves.

Either way at some point we need to consolidate all of this stuff -- see https://github.com/metabase/metabase/issues/27124

**Comment by metamben** [comment]
We've worked around the issue in another way,

**Pull request closed without merging**